File: App_Flutter_rentalhouse/lib/models/favorite.dart
import 'package:flutter_rentalhouse/models/rental.dart';

class Favorite {
  final String userId;
  final String rentalId;
  final Rental? rental;
  final DateTime createdAt;

  Favorite({
    required this.userId,
    required this.rentalId,
    this.rental,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      userId: json['userId'] ?? '',
      rentalId: json['rentalId']?.toString() ?? '',
      rental: json['rentalId'] != null && json['rentalId'] is Map
          ? Rental.fromJson(json['rentalId'] as Map<String, dynamic>)
          : null,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'rentalId': rentalId,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: App-RentHouse/models/favorite.js
const mongoose = require('mongoose');

const favoriteSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
  },
  rentalId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Rental',
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Favorite', favoriteSchema);